crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -183,9 +184,21 @@ async fn download_block(
                 .await
                 .context("Query sequencer for latest block")?;
 
-            if latest.block_number.unwrap() + 1 == block {
-                Ok(DownloadBlock::AtHead)
+            if latest.block_number.unwrap() + 1 == block_number {
+                match prev_block_hash {
+                    // We are definitiely still at the head and it's just that a new block
```

Comment:
Yeah, those mocks are closely hard-wired to the current sync behavior which is pita if you change the behavior. But at the same time since they are so closely coupled you will definitely know if the behavior changed sometimes and how, which is good for finding bugs when trying to improve  the sync process :thinking: